### PR TITLE
Refactor hash merge methods

### PR DESCRIPTION
**Pull request by ghost**
The before and after comparison is very confusing on this one. I'd recommend looking at the changes in lib/games_manager lines 211-246

### CODE REVIEW

File: lib/games_manager.rb [hunk 1]
```diff
@@ -173,91 +173,69 @@ def get_season_games(season)
...
-        visitor_team_id_games[game.away_team_id] = 1
-      else
-        visitor_team_id_games[game.away_team_id] += 1
-      end
+  
+  def sum_values(key_value_arr)
+    sum = Hash.new(0)
+    key_value_arr.each do |key, value|
+      sum[key] += value
     end
```
Comment (ahenzler) [inline lib/games_manager.rb]: really like this helper method.

File: lib/games_manager.rb [hunk 2]
```diff
@@ -173,91 +173,69 @@ def get_season_games(season)
...
+    (averages.min_by {|team_id, average| average})[0]
+  end
+
+  def max_average_hash_values(hash_1, hash_2)
+    averages = hash_1.merge(hash_2) do |key, hash_1_value, hash_2_value|
+      (hash_1_value/hash_2_value.to_f).round(2)
     end
-    average_min[0]
+    (averages.max_by {|team_id, average| average})[0]
   end
```
Comment (ahenzler) [inline lib/games_manager.rb]: I really like how you used helper methods to use hashes.

**Review by ahenzler — COMMENTED** [review]
Brilliant. Really good refactoring, I like the helper methods you created.

File: lib/games_manager.rb [hunk 3]
```diff
@@ -173,91 +173,69 @@ def get_season_games(season)
...
-  def total_home_games
-    home_team_id_games = {}
-    games.each do |game|
-      if home_team_id_games[game.home_team_id].nil?
-        home_team_id_games[game.home_team_id] = 1
-      else
-        home_team_id_games[game.home_team_id] += 1
-      end
+  
+  def total_away_goals
```
Comment (denverdevelopments) [inline lib/games_manager.rb:182]: It looks like Aidan's nice 'map' refactors are from the prior (also was the first) refactor merge.

File: lib/games_manager.rb [hunk 4]
```diff
@@ -173,91 +173,69 @@ def get_season_games(season)
...
+    sum = Hash.new(0)
+    key_value_arr.each do |key, value|
+      sum[key] += value
     end
-    visitor_team_id_games
+    sum
+  end
+  
+  def highest_scoring_home
+    max_average_hash_values(total_home_goals, total_home_games)
```
Comment (denverdevelopments) [inline lib/games_manager.rb:210]: This hash merge helper method is efficient, very nice.

**Review by denverdevelopments — APPROVED** [review]

**Review by Aphilosopher30 — COMMENTED** [review]
everything looks nice and clean.

**Review by ahenzler — APPROVED** [review]

**Pull request closed without merging**